cpp/inc/bond/stream/output_buffer.h

```diff
@@ -165,15 +171,16 @@ class OutputMemoryStream
         return merge(_allocator, merge(_allocator, _blobs.begin(), _blobs.end()), current);
     }
 
-    
+
     template<typename T>
     void Write(const T& value)
     {
+#if defined(__x86_64__) || defined(_M_X64) || defined(__i386) || defined(_M_IX86)
         //
-        // Performance tweak: for small types don't go into generic Write(),
-        // which results in call to memcpy() and additional memory access.
-        // The direct copy of the value (which is likely on register already)
-        // is faster.
+        // x86/x64 performance tweak: for small types don't go into generic
```

Comment:
I re-validated that this tweak still results in better performance--at least with MSVC.